src/jdk.jconsole/share/classes/sun/tools/jconsole/SummaryTab.java

```diff
@@ -257,9 +258,14 @@ synchronized Result formatSummary() {
                     String[] kbStrings1 =
                         formatKByteStrings(sunOSMBean.getCommittedVirtualMemorySize());
 
+                    // getTotalPhysicalMemorySize and getFreePhysicalMemorySize are deprecated,
+                    // but we want be able to get the data for old target VMs (see JDK-8255934).
+                    @SuppressWarnings("deprecation")
                     String[] kbStrings2 =
-                        formatKByteStrings(sunOSMBean.getTotalMemorySize(),
-                                           sunOSMBean.getFreeMemorySize(),
+                        formatKByteStrings(tryToGet(sunOSMBean::getTotalMemorySize,
+                                                    sunOSMBean::getTotalPhysicalMemorySize),
+                                           tryToGet(sunOSMBean::getFreeMemorySize,
+                                                    sunOSMBean::getFreePhysicalMemorySize),
```

Comment:
What happens if/when `getTotalPhysicalMemorySize()` and `getFreePhysicalMemorySize()` are removed. This code will fail to compile, right? I guess at that point the fix would be to try the new API, and if it throws an exception, then just use -1. In other words, there would be no way to get the value from older JVMs.

Reply:
Yes, that's right. If/when the methods are removed, this code will require an update too.
But I think it would be better to show the values as long as the methods exist